component/block_cache/block_cache.go

```diff
@@ -358,7 +358,7 @@ func (bc *BlockCache) OpenFile(options internal.OpenFileOptions) (*handlemap.Han
 			listMap[int64(idx)] = block.Id
 			// All blocks shall of same size otherwise fail the open call
 			// Last block is allowed to be of smaller size as it can be partial block
-			if block.Size != bc.blockSize && idx != (listLen-1) {
+			if (idx < (listLen-1) && block.Size != bc.blockSize) || (idx == (listLen-1) && block.Size > bc.blockSize) {
```

Comment:
Is it possible to add some test case around this

Reply:
Added Unit tests for the code, please review it again